### ISSUE TITLE
False not-an-iterable on subclasses of functools.cached_property

### ISSUE THREAD

**Issue by mharding-hpe** [pylint-dev/pylint#10377]
### Bug description

```python
# pylint: disable=missing-module-docstring,missing-class-docstring,invalid-name,too-few-public-methods,missing-function-docstring

from typing import reveal_type, NoReturn
from functools import cached_property

class cached_property_readonly[T](cached_property[T]):
    def __set__(self, instance: object, val: T) -> NoReturn:
        raise AttributeError("BAD")


class ice_weasels:
    @cached_property
    def harf(self) -> list[int]:
        return [1, 2, 3]

    @cached_property_readonly
    def blah(self) -> list[int]:
        return [1, 2, 3]


d = ice_weasels()
reveal_type(d.harf) # mypy reports list[int]
reveal_type(d.blah) # mypy reports list[int]
for _ in d.harf: # No complaints from mypy or pylint
    pass
for _ in d.blah: # No complaints from mypy, but pylint gives not-an-iterable
    pass

# If the cached_property_readonly class is not Generic over T and inheriting from
# cached_property[T], then mypy reports the revealed type of blah as Any.
```

### Configuration

```ini

```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:26:9: E1133: Non-iterable value d.blah is used in an iterating context (not-an-iterable)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.12.9 (main, Mar 18 2025, 03:28:39) [GCC 12.2.0]

(but I have reproduced on Python 3.10 and 3.13 as well)
```

### OS / Environment

docker.io/python:3.12 x86_64 image

https://mypy-play.net/?mypy=latest&python=3.12&gist=00a0d22bec726e12655bc2ea85c8de5d

### Additional dependencies

```python

```

**Comment by mharding-hpe** [comment]
I should mention that the problem actually reproduces even if I leave the body of the `cached_property_readonly` class empty -- it is not the `__set__` method definition which introduces the problem. Adding that method was my original motivation for making the class, but this `pylint` issue happens even without that present.

**Comment by mharding-hpe** [comment]
I wonder if [this block](https://github.com/pylint-dev/astroid/blob/3636bc299a5792dc2a421cf6951438a396fd9a6e/astroid/bases.py#L93-L99) needs to be updated to include `cached_property`, because I notice that `cached_property` is not a subclass of `property`.

**Comment by mharding-hpe** [comment]
I realize I can work around this by running `pylint` with `--init-hook="import astroid.bases; astroid.bases.POSSIBLE_PROPERTIES.add('cached_property_readonly')"`, but that seems like it's intended for property-like classes that can't be inferred because they don't directly inherit from core Python properties (like `property` or `cached_property`)
